### PR TITLE
docs: add merge guide

### LINKED ISSUE #2915: Add `merge` guide to docs
It would be great to have an extensive merge guide in the docs, incl:
- showcasing each "when statement" :what it does and how you can combine things to do upserts or just insert new records only (de duplicate on insert), and upsert with deleting old records
- showcase when you have multiple of when matched updates, to highlight that the first one that is met is executed

### PR DESCRIPTION
**Pull request by avriiil**
This PR adds a MERGE guide to the docs. Will close #2915 

Signed-off-by: Avril Aysha <68642378+avriiil@users.noreply.github.com>

### CODE REVIEW

File: docs/usage/merging-tables.md [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+> dt.to_pandas().sort_values("x", ignore_index=True)
+   x  y
+0  1  0
+1  2  5
+2  3  6
+```
+
+## Notes
+
+- Column names with special characters, such as numbers or spaces should be encapsulated in backticks: "target.123column" or "target.my column"
```
Comment (ion-elgreco) [inline docs/usage/merging-tables.md:375]: Can you add an example with the backticks?

**Comment by ion-elgreco** [comment]
Looks good! @avriiil

**Comment by avriiil** [comment]
Thanks @ion-elgreco! this is ready for final review.

**Review by ion-elgreco — APPROVED** [review]
Thanks!!

**Pull request merged**